Hang ramdonly when updating VCSStore in unstable network

### Affected Version

`yay -V`: yay v10.1.1 - libalpm v12.0.2
`go version`: go version go1.15.6 linux/amd64

### Describe the bug

When installing some VCS packages (e.g. gnome-shell-extension-shelltile-git), yay randomly hangs during updating the VCSStore. I found that `git ls-remote` takes very long time because of my unstable network and the timout of `func (*OSRunner) Capture(cmd, timeout)` (`pkg/settings/exe/exec.go`) seems not work if child process has its own child process.

### Reproduction Steps

I wrote a unit test to reproduce the bug.

``` go
package exe

import (
	"os/exec"
	"testing"
	"time"
)

func TestCaptureTimeout(t *testing.T) {
	runner := &OSRunner{}

	// cmd := exec.Command("sh", "-c", "watch date > /tmp/date.txt")
	cmd := exec.Command("git", "ls-remote", "https://github.com/yay/yay.git", "HEAD")
	start := time.Now()
	runner.Capture(cmd, 2)

	if time.Since(start) > 3*time.Second {
		t.Fail()
	}
}
```

Then run `go test -timeout 30s github.com/Jguer/yay/v10/pkg/settings/exe -run ^TestCaptureTimeout$ -v -count=1`:

```
=== RUN   TestCaptureTimeout
panic: test timed out after 30s

... some callstack ...

FAIL    github.com/Jguer/yay/v10/pkg/settings/exe       30.010s
FAIL
```

I found a [post](https://medium.com/@felixge/killing-a-child-process-and-all-of-its-children-in-go-54079af94773) seems fix this issue. I'll test on my computer and create a pr if it works.

--- a/pkg/settings/exe/exec.go
+++ b/pkg/settings/exe/exec.go
@@ -6,6 +6,7 @@ import (
 	"os"
 	"os/exec"
 	"strings"
+	"syscall"
 	"time"
 
 	"github.com/Jguer/yay/v10/pkg/text"
@@ -35,6 +36,8 @@ func (r *OSRunner) Capture(cmd *exec.Cmd, timeout int64) (stdout, stderr string,
 
 	cmd.Stdout = &outbuf
 	cmd.Stderr = &errbuf
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
 	err = cmd.Start()
 	if err != nil {
 		stdout = strings.TrimSpace(outbuf.String())
@@ -44,7 +47,7 @@ func (r *OSRunner) Capture(cmd *exec.Cmd, timeout int64) (stdout, stderr string,
 
 	if timeout != 0 {
 		timer = time.AfterFunc(time.Duration(timeout)*time.Second, func() {
-			err = cmd.Process.Kill()
+			err = syscall.Kill(-cmd.Process.Pid, syscall.SIGKILL)
 			if err != nil {
 				text.Errorln(err)
 			}

